InpsydeTemplates/Sniffs/Formatting/AlternativeControlStructureSniff.php

```diff
@@ -0,0 +1,162 @@
+<?php
+
+declare(strict_types=1);
+
+namespace InpsydeTemplates\Sniffs\Formatting;
+
+use PHP_CodeSniffer\Files\File;
+use PHP_CodeSniffer\Sniffs\Sniff;
+use PHP_CodeSniffer\Util\Tokens;
+use PHPCSUtils\Utils\ControlStructures;
+
+/**
+ * The implementation is inspired by Universal.DisallowAlternativeSyntaxSniff.
+ *
+ * @link https://github.com/PHPCSStandards/PHPCSExtra/blob/ed86bb117c340f654eab603a06b95a437ac619c9/Universal/Sniffs/ControlStructures/DisallowAlternativeSyntaxSniff.php
+ *
+ * @psalm-type Token = array{
+ *     type: string,
+ *     code: string|int,
+ *     line: int,
+ *     scope_opener?: int,
+ *     scope_closer?: int,
+ *     scope_condition?: int,
+ *     content: string,
+ *  }
+ */
+final class AlternativeControlStructureSniff implements Sniff
+{
+    /**
+     * @return list<int|string>
+     */
+    public function register(): array
+    {
+        return [
+            T_IF,
+            T_WHILE,
+            T_FOR,
+            T_FOREACH,
+            T_SWITCH,
+        ];
+    }
+
+    /**
+     * @param File $phpcsFile
+     * @param int $stackPtr
+     *
+     * phpcs:disable Inpsyde.CodeQuality.ArgumentTypeDeclaration
+     */
+    public function process(File $phpcsFile, $stackPtr): void
+    {
+        if (ControlStructures::hasBody($phpcsFile, $stackPtr) === false) {
+            // Single line control structure is out of scope.
+            return;
+        }
+
+        /** @var array<int, Token> $tokens */
+        $tokens = $phpcsFile->getTokens();
+        /** @var int | null $scopeOpener */
+        $openerPtr = $tokens[$stackPtr]['scope_opener'] ?? null;
+        /** @var int | null $scopeCloser */
+        $closerPtr = $tokens[$stackPtr]['scope_closer'] ?? null;
+
+        if (!isset($openerPtr, $closerPtr, $tokens[$openerPtr])) {
+            // Inline control structure or parse error.
+            return;
+        }
+
+        if ($tokens[$openerPtr]['code'] === T_COLON) {
+            // Alternative control structure.
+            return;
+        }
+
+        $chainedIssues = $this->findChainedIssues($phpcsFile, $stackPtr);
+
+        $message = 'Control structure having inline HTML should use alternative syntax.'
+            . ' Found "%s".';
+        foreach ($chainedIssues as $conditionPtr) {
+            $phpcsFile->addWarning(
```

Comment:
Looking at `Universal.ControlStructures.AlternativeControlStructureSniff`, I see that it supports auto-fixing. Did you consider this for our sniff as well? Wouldn't this only require replacing the closer (which should always be a closing curly brace `}`) with the correct alternative syntax closing token for the current opener? So we'd be looking at a map with entries `T_IF => T_ENDIF` etc.

Reply:
Yes, I considered making the sniff auto-fixable, as mentioned in the PR description. We also have nuances with openers because it might be two tokens (`else if` - while another sniff forbids it). Also, we have no tests for fixable sniffs. What do you think about the following plan?
1. Get the approval for the current sniffs and create a new issue to make `AlternativeControlStructureSniff` fixable.
2. Create a new PR to improve our testing suite to cover auto-fixing.
3. Create a next PR to make `AlternativeControlStructureSniff` fixable.

Reply:
Personally, I'm fine with merging as is now. Improving tests and adding auto-fixing is something we may want to do on the new PHP Coding Standards, and not here... I only wanted to ask about the missing auto-fixing, while it is included in the "base sniff". I did not remember this being mentioned in the PR description, which I read last week, sorry.